dsc/sqs.py

```diff
@@ -0,0 +1,203 @@
+                    logger.debug(
+                        "Message retrieved from SQS queue %s: %s",
+                        self.queue_name,
+                        message,
+                    )
+                    yield message
+            else:
+                logger.debug("No more messages from SQS queue: %s", self.queue_name)
+                break
+
+    def send(
+        self,
+        message_attributes: Mapping[str, MessageAttributeValueTypeDef],
+        message_body: str,
+    ) -> SendMessageResultTypeDef:
+        """Send message via SQS.
+
+        Args:
+            message_attributes: The attributes of the message to send.
+            message_body: The body of the message to send.
+        """
+        logger.debug("Sending message to SQS queue: %s", self.queue_name)
+        response = self.client.send_message(
+            QueueUrl=f"{self.base_url}{self.queue_name}",
+            MessageAttributes=message_attributes,
+            MessageBody=str(message_body),
+        )
+        logger.debug("Response from SQS queue: %s", response)
+        return response
+
+    @staticmethod
+    def valid_result_message_attributes(sqs_message: MessageTypeDef) -> bool:
+        """Validate that "MessageAttributes" field is formatted as expected.
+
+        Args:
+            sqs_message: An SQS message to be evaluated.
+        """
+        valid = False
+        if (
+            "MessageAttributes" in sqs_message
+            and any(
+                field
+                for field in sqs_message["MessageAttributes"]
+                if "PackageID" in field
+            )
+            and sqs_message["MessageAttributes"]["PackageID"].get("StringValue")
+        ):
+            valid = True
+        else:
+            logger.exception("Failed to parse SQS message attributes: %s", sqs_message)
+        return valid
+
+    @staticmethod
+    def valid_result_message_body(sqs_message: MessageTypeDef) -> bool:
+        """Validate that "Body" field is formatted as expected.
+
+        Args:
+            sqs_message: An SQS message to be evaluated.
+        """
+        valid = False
+        if "Body" in sqs_message and json.loads(str(sqs_message["Body"])):
+            valid = True
+        else:
+            logger.exception("Failed to parse SQS message body: %s", sqs_message)
+        return valid
+
+    def valid_sqs_message(self, sqs_message: MessageTypeDef) -> bool:
+        """Validate that an SQS message is formatted as expected.
+
+        Args:
+            sqs_message:  An SQS message to be evaluated.
+
+        """
+        valid = False
+        if (
+            self.valid_result_message_attributes(sqs_message=sqs_message)
+            and self.valid_result_message_body(sqs_message=sqs_message)
+            and sqs_message.get("ReceiptHandle")
+        ):
+            valid = True
+        else:
+            logger.exception("Failed to parse SQS message: %s", sqs_message)
```

Comment:
Going to pick on this `logger.exception` line, but I think it may apply more broadly to the flow here, and the logging messages from the helper methods.

Let's say that `self.valid_result_message_body()` cannot load the `Body` part of the message as JSON.  Wouldn't both that method, and this method `valid_sqs_message()` (or `validate_sqs_message()` if renamed) both log the full SQS message?  Something like:

```
Failed to parse SQS message body: <SQS message here...>
Failed to parse SQS message: <SQS message here...>
```

To me what feels ideal would be logging like:

```
"Invalid SQS message: failed to parse message body
```

Where the first part is coming from the `validate_sqs_message()` method, but the _reason_ "failed to parse message Body as JSON" came from the helper method `valid_result_message_body`.

It's all pretty minor, and frankly not blocking for my approval, but feels like maybe those helper methods that are doing the validation work could just return tuples like `(<success:bool>, <reason:str>)` that this `validate_sqs_message()` method could use.

Reply:
I should have led with: I do like this validation approach!  It's nice they are in standalone methods.  

I'd be okay with the approach as-is, there are things I like about it too.  Please take all comments here as optional / conversational, but by no means requested changes.

Reply:
I did refactor this method to prevent dupe logging, let me know how it looks

Reply:
@ehanson8 - maybe missing it somehow, but not seeing a new commit?

Reply:
Just pushed!